src/RiddleFeedService.php

```diff
@@ -157,17 +165,58 @@ private function processRiddleResponse($riddleResponse) {
           continue;
         }
 
-        $feed[] = array(
-          'title' => $this->getRiddleTitle($riddleEntry),
-          'uid' => $riddleEntry['uid'],
-        );
+        $image = NULL;
+        if (!empty($riddleEntry['data']['image']['standard'])) {
+          $image = $this->getImage($riddleEntry['data']);
+        }
+        elseif (!empty($riddleEntry['draftData']['image']['standard'])) {
+          $image = $this->getImage($riddleEntry['draftData']);
+        }
+
+        if ($this->fetchUnpublished || $riddleEntry['status'] == 'published') {
```

Comment:
It would be better to move this filtering before $image evaluation. And it can be same as ```if (!$this->isValidRiddleFeedEntry($riddleEntry)) {``` -> just continue if it should be skipped.